src/dafny/Relations.md

```diff
@@ -0,0 +1,36 @@
+
+## The `Relations` module {#sec-relations}
+
+The `Relations` module defines a number of properties of functions.
+
+Properties of binary predicates:
+- Reflexive: `R(x,x)` is true
+- Irreflexive: `R(x,x)` is false
+- AntiSymmetric: `R(x,y) && R(y,x) ==> x==y`
+- Transitive : `R(x,y) && R(y,z) ==> R(x,z)`
+- Connected : `x != y ==> R(x,y) || R(y,x)`
+- StronglyConnected : `R(x,y) || R(y,x)`
+- TotalOrdering : Reflexive, AntiSymmetric, Transitive, StronglyConnected (e.g., `<=` on integers)
+- StrictTotalOrdering : Irreflexive, AntiSymmetric, Transitive, Connected (e.g., `<` on integers)
+
+A property of unary functions:
+- Injective : `f(x) == f(y) ==> x == y`
+
+These properties are sometimes required for other functions. For example, 
+if one wants to sort a sequence by some relation `R`, one must establish that `R` is a _Total Ordering_. 
+In fact, that is part of the precondition of the sorting function.
+
+As a simple example, you might define a predicate like this:
+```dafny
+  const IntLT  := ((i: int, j: int) => (i < j))
+```
+
+and then need to proof this lemma to use it in a sorting routine:
+```dafny
+  lemma IntLTisStrictTotalOrder()
+    ensures StrictTotalOrdering(IntLT) {}
+```
```

Comment:
Again better to only have under examples - perhaps add the content here as comments there instead?

Reply:
Other reviewers (of other documentation) have predominantly preferred having examples directly present, both for reading and reviewing. I'll add the annotation to be sure these are checked in place.

Reply:
Sure, as above I could see steering this repo towards having all inline examples instead of the current setup, and IF it is possible to set up the mechanism to test them I'm happy with that.

Reply:
It is possible -- just like it is already in the other repo. But it has to be checked in and next to the new code.
Then I'll be able to create a yml to do the checking regularly or on demand.